### PR TITLE
Add JSON mapping of ship name to character

### LINKED ISSUE #43: Add JSON mapping of ship name to character
Please add a JSON file that contains a mapping of ship name to icon. It should look something like this:
```
{
	"aggressor": "i",
	"alphaclassstarwing": "&",
	"arc170": "c",
	"attackshuttle": "g",
        ...
        "ywing": "y",
	"z95headhunter": "z"
}
```

I maintain the iOS app [DialVision](https://itunes.apple.com/us/app/dialvision/id1277244707) which displays a list of ships and their icons. The app dynamically loads xwing-miniatures-ships.ttf from GitHub to ensure that it will display icons for the latest ships without having to redistribute the app every time the font updates but it needs a JSON file that maps the ship name to each character so it knows what character icon to display for each ship. Currently, I have a [fork of this repo](https://github.com/mludowise/xwing-miniatures-font/tree/json-map) and have been manually maintaining a JSON file but it would be great if I didn't have to maintain a fork and manually generate the JSON every time there's an update.

Here's a screenshot from my app of the view I'm talking about:
Screenshot

I'd love to help in any way I can, but I'm not sure what the best approach would be. I'm not totally familiar with your build process, but I'm guessing the solution could involve using JSON as the source and generating the *-map.scss files as part of the Grunt task.

### PR DESCRIPTION
**Pull request by mludowise**
PR to resolve #43.

Uses JSON files as source files for icon maps and generates _*-maps.scss files and index.html files from JSON using Handlears templates.

### CODE REVIEW

File: Gruntfile.coffee [hunk 1]
```diff
@@ -1,5 +1,40 @@
 module.exports = (grunt) ->
   grunt.initConfig
+    
+    # Delete files that will be used by compile-handlebars because it appends to the end of a file rather than re-writing it.
+    clean: [ 'src/sass/_*-map.scss', 'index.html']
```
Comment (geordanr) [inline Gruntfile.coffee]: Might as well remove `_*-map.scss` from git if they're only going to be used for generating CSS.  That way we aren't tracking as much churn when the source JSON changes.

File: handlebars/helpers/escape.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module.exports = function(variable) {
+    // return variable.replace(/(['"])/g, '\\$1');
+    
+    return variable.replace(/[\s\S]/g, function(character) {
```
Comment (geordanr) [inline handlebars/helpers/escape.js]: Why `/[\s\S]/g` instead of `/./g`?

File: Gruntfile.coffee (hunk 1 above)
Comment (mludowise) [reply Gruntfile.coffee]: Might they be needed for anyone using SCSS rather than the compiled CSS?

**Comment by mludowise** [comment]
Yes, these changes would deprecate Pug. If you prefer to continue using Pug to generate index.html, it shouldn't be too hard to update it to use the JSON. My thinking was that it made more sense to only use one templating language, but I don't feel very strongly about it.

File: handlebars/helpers/escape.js (hunk 2 above)
Comment (mludowise) [reply handlebars/helpers/escape.js]: Good catch. I missed that when copying this over from StackOverflow.

File: Gruntfile.coffee (hunk 1 above)
Comment (geordanr) [reply Gruntfile.coffee]: That's true, though I'm not sure if anyone's actually using that.  Safe to leave in for now.

**Comment by geordanr** [comment]
I'm fine with deprecating Pug.

**Comment by mludowise** [comment]
I updated escape.js with your suggestion.

**Comment by geordanr** [comment]
Awesome!

**Pull request merged**